zerver/tests/test_subs.py

```diff
@@ -5935,7 +5969,8 @@ def test_inviteonly(self) -> None:
 
         # Subscribing oneself to an invite-only stream is not allowed
         self.login_user(othello)
-        result = self.common_subscribe_to_streams(othello, [stream_name], allow_fail=True)
+        with transaction.atomic():
+            result = self.common_subscribe_to_streams(othello, [stream_name], allow_fail=True)
```

Comment:
Why are these changes needed? First, if we need a transaction here, it should go in `self.common_subscribe_to_streams`, not every caller. But I also don't understand why it would be required at all.

Reply:
Ahh, I see the commit message comments on it. Post in #backend; there's probably a better way to do this, but also we should do this in `self.common_subscribe_to_streams` if it's required.

Reply:
Updated the PR to include the context manager in `self.common_subscribe_to_streams`.

Also a prep commit (3rd commit) to use the `self.common_subscribe_to_streams` instead of independent API calls. (Helps to avoid adding the context manager independently to those API calls)